### DISCUSSION TITLE
[Config Support]: EXIF data not in clips or snapshots

### DISCUSSION THREAD

**Discussion by aburow** [blakeblackshear/frigate#10876] [Config Support]
### Describe the problem you are having

I am not able to figure out how to put basic exif data into snapshots and recordings in order to preserve simple information, such as "time/date acquired" field. It would be nice to be able to fill out as much data as possible but I can't find a way to do this.

### Version

0.13.2-6476F8A

### Frigate config file

```yaml
logger:
  default: warning
  logs:
    frigate.mqtt: error

ui:
  timezone: Australia/Brisbane
  strftime_fmt: "%H:%M %d/%m/%Y"
  live_mode: mse

mqtt:
  enabled: true
  user: ""
  password: ""
  host: 192.168.100.10
  stats_interval: 60

detectors:
 coral:
   type: edgetpu
   device: usb

detect:
  stationary:
    interval: 10
    threshold: 50

audio:
  enabled: true

ffmpeg:
  hwaccel_args: preset-vaapi
  output_args:
    record: preset-record-generic-audio-copy
  global_args:
    -hide_banner
    -loglevel info

birdseye:
  enabled: True
  restream: false
  mode: objects

record:
  enabled: true
  retain:
    days: 5
    mode: motion
  events:
    pre_capture: 10
    post_capture: 30
    retain:
      default: 5
      mode: active_objects

snapshots:
  enabled:  True
  timestamp: True
  bounding_box: True
  retain:
    default: 5

go2rtc:
  streams:
    floorright:
      # Dahua
      - "rtsp://100.66.67.101:554#backchannel=0"
    garage:
      # XM510
      - "rtsp://100.66.67.103/channel=1_stream=0.sdp"
    bicam:
      # Hik
      - "rtsp://100.66.67.105:554/stream0#backchannel=0"
    ballcam:
      # Hik
      - "rtsp://100.66.67.106:554/stream0#backchannel=0"

cameras:
  floorright:
    ui:
      order: 1
    audio:
      enabled: false
    ffmpeg:
      inputs:
        - path: "rtsp://127.0.0.1:8554/floorright"
          input_args: preset-rtsp-restream
          roles:
            - detect
            - record
    detect:
      enabled: True
      fps: 10
    objects:
      track:
        - person
        - bird
        - cat
        - dog
        - horse
        - sheep
        - cow
        - bear
      filters:
        person:
          threshold: 0.8
        bird:
          min_score: 0.7
          min_area: 1000
        cat:
          threshold: 0.4
        dog:
          threshold: 0.4

  garage:
    ui:
      order: 2
    enabled: true
    ffmpeg:
      inputs:
        - path: "rtsp://127.0.0.1:8554/garage"
          input_args: preset-rtsp-restream
          roles:
            - audio
            - record
            - detect
    audio:
      enabled: true
      max_not_heard: 30
      min_volume: 500
      listen:
        - bark
      filters:
        bark:
          threshold: 0.5
    detect:
      enabled: True
      fps: 10
    objects:
      track:
        - person
        - bird
        - cat
        - dog
        - horse
        - sheep
        - cow
        - elephant
        - bear
      filters:
        person:
          threshold: 0.8
        cat:
          threshold: 0.4
        dog:
          threshold: 0.4

  bicam: #Name for your comment
    ui:
      order: 3
    audio:
      enabled: false
    ffmpeg:
      inputs:
        - path: "rtsp://127.0.0.1:8554/bicam"
          input_args: preset-rtsp-restream
          roles:
            - detect
            - record
    detect:
      enabled: True
      fps: 20
    objects:
      track:
        - person
        - bird
        - cat
        - dog
        - horse
        - sheep
        - cow
        - elephant
        - bear
      filters:
        person:
          threshold: 0.8
        bird:
          min_score: 0.7
          min_area: 4000
        cat:
          threshold: 0.4
        dog:
          threshold: 0.4

  ballcam: #Name for your comment
    ui:
      order: 4
    audio:
      enabled: false
    ffmpeg:
      inputs:
        - path: "rtsp://127.0.0.1:8554/ballcam"
          input_args: preset-rtsp-restream
          roles:
            - detect
            - record
    detect:
      enabled: True
      fps: 15
    objects:
      track:
        - person
        - bird
        - cat
        - dog
        - horse
        - sheep
        - cow
        - elephant
        - bear
      filters:
        person:
          threshold: 0.8
        cat:
          threshold: 0.4
        dog:
          threshold: 0.4

#EOF
```

### Relevant log output

```shell
n/a
```

### Operating system

Proxmox

### Install method

HassOS Addon

### Coral version

USB

**Comment by NickM-27** [comment]
what are you using to add exif data and what are you hoping to use that data for?

**Reply by aburow** [reply]
Metadata is embedded within security images and videos so that the information can stand-alone as evidence. At the moment this data does appear to be stored in the database, however, this data is separated/readily severable from the photographic evidence created by Frigate itself.

When reviewing data for forensic purposes information directly embedded into the datasource at the source appears to be the preferred option, having the file digitally signed is a bonus. Using multiple sources (database + photo filename + directory structure in this case)  is problematic as the image/video is lifted out of that context and may even have been renamed during that process. There can be additional questions about the validity of the data and integrity of the process used to assemble the dataset. I am fully aware that I can just use tools that can edit exif and xmp data nullifying some of the benefit, but having this in combination with something such as an openssl signature on the end will assist in covering that issue. Some of the security camera/DVR/NVR systems I've worked with previously provide additional schemes to ensure integrity of the evidence isn't broken once the file leaves the originating system.

This information I have provided goes well beyond the metadata that I would seek at this stage which amounts to providing some integrity around the date and time the photo/video was recorded. It would be nice to be able to fill in exif/xmp information such as lat/lon, cameraname, events, etc as fields for later review and management away from Frigate.

TL;DR - photo and video data taken from frigate loses context including time, date, location. Filesystem data is unreliable. Frigate is already creating event tags. XMP and OpenSSL offer metadata and integrity options.

**Reply by aburow** [reply]
Found this entry in stackoverflow:

https://stackoverflow.com/questions/67068242/how-to-add-arbitrary-or-custom-metadata-in-mp4

Describes the use of metadata on MP4 files.
